be/src/olap/reader.cpp

```diff
@@ -458,7 +458,8 @@ OLAPStatus Reader::_init_params(const ReaderParams& read_params) {
 OLAPStatus Reader::_init_return_columns(const ReaderParams& read_params) {
     if (read_params.reader_type == READER_QUERY) {
         _return_columns = read_params.return_columns;
-        if (!_delete_handler.empty() && read_params.aggregation) {
```

Comment:
For reviewer:
Remove `read_params.aggregation` because this is just a tricky optimization.
That is, if `aggregation` is false, than all key columns has already been added to the `_return_columns`.
And delete conditions can only be applied on key columns, so no need to iterate and add them again.

But this is not always right, because for DUPLICATION KEY table, the delete conditions can be applied on any columns. Although the query optimizer will guarantee that for  DUPLICATION KEY table, the aggregation is always true, but we should not depend on it.